### PR TITLE
Cleanup: contract build, old tutorials etc.

### PR DESCRIPTION
**Pull request by andreibancioiu**
#### Description of the pull request (what is new / what has changed)

 - Use `sc-meta` instead of `mxpy` for contract-related tasks.
 - Remove tutorial on `Simple Counter` (C contracts cannot be built anymore with _mxpy_ since ~2023). Rust is recommended.

#### Did you test the changes locally ?
- [x] yes
- [ ] no

#### Which category (categories) does this pull request belong to?
- [ ] document new feature
- [x] update documentation that is not relevant anymore
- [ ] add examples or more information about a component
- [ ] fix grammar issues
- [ ] other

### CODE REVIEW

**Review by radumojic — DISMISSED** [review]

**Review by danidrasovean — APPROVED** [review]

**Review by popenta — DISMISSED** [review]

File: docs/developers/meta/sc-build-reference.md [hunk 1]
```diff
@@ -157,8 +151,6 @@ After building the contracts, there are three more operations left to perform, b
 
 Calling `cargo run clean` in the meta crate will run `cargo clean` in all wasm crates and delete the `output` folder.
```
Comment (BiancaIalangi) [inline docs/developers/meta/sc-build-reference.md:152]: Clean is also available using: `sc-meta all clean`

File: docs/developers/meta/sc-meta-cli.md [hunk 2]
```diff
@@ -388,7 +388,7 @@ This command is similar to `build-dbg`, in that it provides a shorthand for buil
 
 ### Calling `clean`
 
-Calling `mxpy contract clean <project>` or `cargo run clean` in the meta crate will delete the `output` folder and clean outputs of the Rust crates.
+Calling `cargo run clean` in the meta crate will delete the `output` folder and clean outputs of the Rust crates.
```
Comment (BiancaIalangi) [inline docs/developers/meta/sc-meta-cli.md:391]: `sc-meta all clean` is used in the contract folder

**Review by BiancaIalangi — CHANGES_REQUESTED** [review]

File: docs/developers/meta/sc-build-reference.md (hunk 1 above)
Comment (andreibancioiu) [reply docs/developers/meta/sc-build-reference.md]: Fixed.

File: docs/developers/meta/sc-meta-cli.md (hunk 2 above)
Comment (andreibancioiu) [reply docs/developers/meta/sc-meta-cli.md]: Fixed.

**Review by BiancaIalangi — APPROVED** [review]

**Review by popenta — APPROVED** [review]

**Pull request merged**